### PR TITLE
Upgraded and refactored Traefik

### PR DESCRIPTION
**Pull request by iBobik**
I wrote routing rules only for services I use for now.

Is it ok like this?

### CODE REVIEW

File: docker-compose-generator/docker-fragments/traefik.yml [hunk 1]
```diff
@@ -3,23 +3,19 @@ version: "3"
 services:
   traefik:
     restart: unless-stopped
-    image: traefik
+    image: traefik:v2.5
     container_name: traefik
     ports:
       - "${REVERSEPROXY_HTTP_PORT:-80}:80"
       - "${REVERSEPROXY_HTTPS_PORT:-443}:443"
+      - "8080:8080" # Dashboard, enable for debug only
```
Comment (Kukks) [inline docker-compose-generator/docker-fragments/traefik.yml:11]: Remove or find a way to protect

**Review by Kukks — CHANGES_REQUESTED** [review]
It seems generally good. Thank you so much and sorry for the long delay in reviewing!

* Could you add the additional routes?
* Can we handle the tor integration too somehow?

File: docker-compose-generator/docker-fragments/traefik.yml (hunk 1 above)
Comment (iBobik) [reply docker-compose-generator/docker-fragments/traefik.yml:11]: It is disabled in traefik.yml by default, you need to uncomment it to enable dashboard.

**Comment by iBobik** [comment]
I do not use Tor, is there somebody who use Tor and has will to experiment with Traefik?

**Comment by Kukks** [comment]
> I do not use Tor, is there somebody who use Tor and has will to experiment with Traefik?

let's do it another time then.

**Review by Kukks — APPROVED** [review]

**Comment by Kukks** [comment]
Is there a reason this is still a draft @iBobik

**Comment by iBobik** [comment]
> Is there a reason this is still a draft @iBobik

No, if you are ok with only some services supported. How about to leaving note in a docs like this?
- there are routing rules missing for some services (`traefik.http.routers` labels, PRs welcome)
- TOR support not tested
- when using BTCPAY_ADDITIONAL_HOSTS don't forgot to also add routing rule, see Traefik/README.md

Or maintain all this caveats in just Traefik/README.md and in BTCPay's docs have just link to it?

**Comment by Kukks** [comment]
> > Is there a reason this is still a draft @iBobik
> 
> No, if you are ok with only some services supported. How about to leaving note in a docs like this?
> 
> * there are routing rules missing for some services (`traefik.http.routers` labels, PRs welcome)
> * TOR support not tested
> * when using BTCPAY_ADDITIONAL_HOSTS don't forgot to also add routing rule, see Traefik/README.md
> 
> Or maintain all this caveats in just Traefik/README.md and in BTCPay's docs have just link to it?

ACK

**Comment by Kukks** [comment]
Will merge once that's done. Thanks again for this contrib! :)

**Comment by NicolasDorier** [comment]
@iBobik I needed to revert this on https://github.com/btcpayserver/btcpayserver-docker/commit/9d0edbfa12d9ecb3710cbb011fb6258949af151e because it was bricking some installs with this error during update:

```
ERROR: The Compose file '/root/btcpayserver-docker/Generated/docker-compose.generated.yml' is invalid because:
services.bitcoin_rtl.labels.traefik.enable contains true, which is an invalid type, it should be a string, number, or a null
services.btcpayserver.labels.traefik.enable contains true, which is an invalid type, it should be a string, number, or a null
```

I think changing the boolean to 'true' might work. But I couldn't test.

**Comment by NicolasDorier** [comment]
@iBobik can you just put those labels in `traefik.yml` instead.

Those settings will get then merged by our docker-compose builder in the resulting docker-compose.
Installations not using traeffik won't have the labels.

**Comment by NicolasDorier** [comment]
I attempt doing it on https://github.com/btcpayserver/btcpayserver-docker/commit/f4d7a12488ef5d4c3b83bb069251f536aa34035c I don't know if it works, so please test.

**Comment by iBobik** [comment]
Reason why it is in every service is to have it on eyes for people changing services yamls. It is easy to forgot something they does not use and to break it.

Btw, the same with Nginx env variables.

So I prefer to fix this incompatibility, which we have to do anyway.

For people with bricked instances, do they use some older Docker compose? I have no other idea why it works for most users.

Honza Pobořil
💬Signal

**Comment by iBobik** [comment]
@NicolasDorier Could we talk about this please? I do not agree with your solution.

> Reason why it is in every service is to have it on eyes for people changing services yamls. It is easy to forgot something they does not use and to break it. Btw, the same with Nginx env variables. So I prefer to fix this incompatibility, which we have to do anyway. For people with bricked instances, do they use some older Docker compose? I have no other idea why it works for most users.

**Pull request merged**